servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingHttpPayloadHolder.java

```diff
@@ -135,8 +135,9 @@ void transformMessageBody(UnaryOperator<Publisher<?>> transformer) {
 
     <T> void transform(final TrailersTransformer<T, Buffer> trailersTransformer) {
         if (messageBody == null) {
-            messageBody = from(trailersTransformer.payloadComplete(trailersTransformer.newState(),
-                    headersFactory.newEmptyTrailers()));
+            messageBody = defer(() ->
+                from(trailersTransformer.payloadComplete(trailersTransformer.newState(),
+                        headersFactory.newEmptyTrailers())));
```

Comment:
This change is not related to the described issue. Can you please open a separate PR with a test for this use-case?

Reply:
Issue: https://github.com/apple/servicetalk/issues/1576
I will spend some more time on that and follow with a PR if needed.